packages/oui-pagination/src/index.spec.js

```diff
@@ -0,0 +1,336 @@
+                        on-change="$ctrl.clickHandler($event)">
+                    </oui-pagination>
+                `, {
+                        pageSize: 25,
+                        clickHandler: clickSpy
+                    });
+
+                const pageSizesButtons = getProgress(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                const contextController = element.scope().$ctrl;
+
+                angular.element(pageSizesButtons[1]).triggerHandler("click");
+                expect(clickSpy).toHaveBeenCalledWith(jasmine.objectContaining({ pageSize: 50 }));
+                expect(clickSpy.calls.count()).toEqual(1);
+
+                // Set the new page size (one-way)
+                contextController.pageSize = 50;
+                element.scope().$digest();
+                expect(pageSizesButtons[1].hasAttribute("disabled")).toBeTruthy();
+
+                // Count page buttons
+                const pageButtons = getSelector(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                expect(pageButtons.length).toEqual(20);
+            });
+
+            it("should have max page size limiting the page size list (without creating a new value in this list)", () => {
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="25"
+                        page-size-max="50"
+                        total-items="1000">
+                    </oui-pagination>
+                `);
+
+                const pageSizesButtons = getProgress(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                expect(pageSizesButtons.length).toEqual(2);
+            });
+
+            it("should have max page size limiting the page size list (and creating a new value in this list)", () => {
+                const element = TestUtils.compileTemplate(`
+                        <oui-pagination
+                            current-offset="1"
+                            page-size="25"
+                            page-size-max="$ctrl.pageSizeMax"
+                            total-items="1000">
+                        </oui-pagination>
+                    `, {
+                        pageSizeMax: 80
+                    });
+
+                let pageSizesButtons = getPageSizeButtons(element);
+                const contextController = element.scope().$ctrl;
+
+                expect(pageSizesButtons.length).toEqual(3);
+                expect(pageSizesButtons[2].innerHTML).toEqual("80");
+
+                // Reset to all page sizes
+                delete contextController.pageSizeMax;
+                element.scope().$digest();
+                pageSizesButtons = getPageSizeButtons(element);
+                expect(pageSizesButtons.length).toEqual(4);
```

Comment:
Not easy to understand, pageSizeMax is forced to 80, after it's deleted and page size 200 is now available.

Reply:
Updated, I hope better.